openbb_platform/core/pyproject.toml

```diff
@@ -9,17 +9,18 @@ packages = [{ include = "openbb_core" }]
 
 [tool.poetry.dependencies]
 python = "^3.9"
-uvicorn = "^0.24"
-websockets = "^12.0"
+uvicorn = "^0.32.0"
+websockets = "^13.0"
+starlette = ">=0.40.0"
```

Comment:
IIUC the fastapi bump resolves the starlette issue. If that's the case there is no reason to pin starlette independently as it's not a top level dependency we use.

Reply:
FastAPI defines Starlette as (https://github.com/fastapi/fastapi/blob/master/pyproject.toml):

```
dependencies = [
    "starlette>=0.37.2,<0.41.0",
    "pydantic>=1.7.4,!=1.8,!=1.8.1,!=2.0.0,!=2.0.1,!=2.1.0,<3.0.0",
    "typing-extensions>=4.8.0",
]
```

So in order for Poetry to not use the lower versions, we need to specify it directly in our package..

Reply:
No need to pin starlette. With unpinned starlette openbb-core already installs 0.40.0
Also this PR https://github.com/fastapi/fastapi/pull/12469 will likely bump the minimal supported version on FastAPI level

Reply:
If you inspect `poetry.lock`, it will still contain the lower version because the latest version of FastAPI simply "allows" a version higher than 0.39 to install.

So until they make a new release of FastAPI, we need to do this. Maybe we just need to wait it out and next week a patch will get published?

Reply:
the poetry.lock of the core has 0.40.0
<img width="710" alt="image" src="https://github.com/user-attachments/assets/cc880f23-2c83-4409-b0f8-be283c1ad603">
IIUC all other lock files have older versions because the core has not ben published or explicitly updated thus they rely on an older fastapi which pulls the old starlette
<img width="303" alt="image" src="https://github.com/user-attachments/assets/b5d69cfa-3e53-48a3-ab04-6a3a3153ea9d">
Once we update the core, the minimal version of fastapi will be bumped and this will bump starlette

Reply:
I propose doing this in a separate PR limiting the scope of this one to only openbb-core